Source/WebCore/loader/NavigationScheduler.cpp

```diff
@@ -606,11 +609,35 @@ void NavigationScheduler::scheduleLocationChange(Document& initiatingDocument, S
         return completionHandler(ScheduleLocationChangeResult::Completed);
     }
 
+    // Fire Navigation API navigate event synchronously before scheduling navigation.
+    // This ensures proper event ordering where navigate event fires before microtasks.
+    // Only fire for same-origin navigations to avoid cross-origin issues.
+    bool hasDispatchedNavigateEvent = false;
+    if (localFrame && !url.protocolIsJavaScript()) {
+        RefPtr document = localFrame->document();
+        if (document && document->settings().navigationAPIEnabled()) {
+            bool isSameOrigin = document->securityOrigin().isSameOriginAs(securityOrigin);
```

Comment:
This result having a variable name isn't useful.  It could just be in the if statement above:
if (document && document->settings().navigationAPIEnabled() && document->securityOrigin().isSameOriginAs(securityOrigin)) {